### PR TITLE
HS-1111: MP Alert Conditions

### PR DESCRIPTION
**Pull request by mikewrosey**
## Description
Handles alert conditions for both threshold and events

screenshot-localhost_3009-2023 03 24-12_39_48

## Jira link(s)
- https://issues.opennms.org/browse/HS-1111

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesCard.vue [hunk 1]
```diff
@@ -35,6 +38,7 @@
         <div class="alert-title">Alert Conditions</div>
         <MonitoringPoliciesCardAlertRow
           v-for="(condition, index) in rule.conditions"
```
Comment (chinh-cle) [inline ui/src/components/MonitoringPolicies/MonitoringPoliciesCard.vue:40]: missing `key` binding attribute

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesPolicyForm.vue [hunk 2]
```diff
@@ -53,8 +52,9 @@
         class="mp-card-container"
       >
         <MonitoringPoliciesCard
-          v-for="policy in store.monitoringPolicies"
+          v-for="(policy, index) in store.monitoringPolicies"
```
Comment (chinh-cle) [inline ui/src/components/MonitoringPolicies/MonitoringPoliciesPolicyForm.vue:55]: missing key binding attribute

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue [hunk 3]
```diff
@@ -77,12 +78,31 @@
           v-if="store.selectedRule!.conditions.length"
         >
           <div class="col">
-            <div class="subtitle">Set Alert Conditions</div>
+            <div class="form-title">Set Alert Conditions</div>
             <MonitoringPoliciesThresholdCondition
-              v-for="cond in store.selectedRule!.conditions"
-              :condition="cond"
+              v-if="store.selectedRule!.detectionMethod === DetectionMethodTypes.Threshold"
+              v-for="(cond, index) in store.selectedRule!.conditions"
```
Comment (chinh-cle) [inline ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue:84]: I think it's not recommended to have v-if and v-for in same element

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue [hunk 4]
```diff
@@ -77,12 +78,31 @@
...
+              :index="index"
+              :condition="(cond as ThresholdCondition)"
```
Comment (chinh-cle) [inline ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue:86]: missing :key attribute for v-for

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue [hunk 5]
```diff
@@ -77,12 +78,31 @@
...
               @updateCondition="(condition) => store.updateCondition(cond.id, condition)"
+              @deleteCondition="(id: string) => store.deleteCondition(id)"
             />
+            <MonitoringPoliciesEventCondition
+              v-else
+              v-for="(cond, index) in store.selectedRule!.conditions"
+              :condition="(cond as EventCondition)"
+              :index="index"
```
Comment (chinh-cle) [inline ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue:94]: same comment as above

**Review by chinh-cle — CHANGES_REQUESTED** [review]

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesPolicyForm.vue (hunk 2 above)
Comment (mikewrosey) [reply ui/src/components/MonitoringPolicies/MonitoringPoliciesPolicyForm.vue:55]: done!

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue (hunk 3 above)
Comment (mikewrosey) [reply ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue]: you're right, wrapped them in template elements

**Review by chinh-cle — APPROVED** [review]
LGTM

**Pull request merged**